src/android/DBHelper.java

```diff
@@ -4,23 +4,45 @@
 import android.database.Cursor;
 import android.database.sqlite.SQLiteDatabase;
 import android.database.sqlite.SQLiteOpenHelper;
+import android.util.Log;
 
+import org.apache.http.client.protocol.ClientContextConfigurer;
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
 
-/**
- * Created by Slonic on 1/26/2016.
- */
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+
+
 public class DBHelper extends SQLiteOpenHelper {
 
+    private int g_version;
+    private static DBHelper sInstance;
+    private static SQLiteDatabase mDatabase;
+
+
+    public static synchronized DBHelper getInstance(Context context, String name, SQLiteDatabase.CursorFactory factory, int version){
+        if (sInstance == null){
```

Comment:
I see what you want to do here, but it won't work the way you intended: 

The first call to `new DBHelper(...)` creates a DBHelper instance, and every subsequent call to `getInstance` returns that very same instance. So when you [do this](https://github.com/miraculixx/cordova-rfscanner-plugin/pull/6/files#diff-f99c396f49972395a9834434bd5a19c5R115), while all DBs are created, all three calls will return the same sInstance.... As a result, any DB insert will go to the first db created and will probably fail... In combination with some insert statements being unsafe (using `insert into table values(...)` instead of `insert into table(columns) values(....)` we can't be sure what gets actually stored and where.

What you need to do is to create one per db `name` (as you do), and then return the respective instance in getInstance, instead of just the first one created. One way to achieve this is by a map  `<String>:DBHelper`.